service/qos/service_level_controller.cc

```diff
@@ -281,6 +281,14 @@ future<> service_level_controller::update_effective_service_levels_cache() {
     });
 }
 
+future<> service_level_controller::update_cache(update_only_effective_cache update_only_effective_cache) {
+    assert(this_shard_id() == global_controller);
+    if (!update_only_effective_cache) {
+        co_await update_service_levels_cache();
+    }
+    co_await update_effective_service_levels_cache();
```

Comment:
These will take the `notifications_serializer` semaphore twice. Can something break if a rogue `add_service_level` executes between them?

Reply:
No, it's not a problem.
`add_service_level` adds the entry to service level cache, so it's up-to-date.